### DISCUSSION TITLE
How to display groups correctly in grid files

### DISCUSSION THREAD

**Discussion by siwux** [neperfepx/neper#322] [Q&A] [answered]
Hi there, many thanks for providing such an amazing tool with NEPER!
I was wondering how to display groups correctly in grid files and enter it into INP files. I also want to know how to use "-elset"
The model built in module T can see grouping correctly in module V
```
neper -T -n 100 -group  "id<=50?1:2"  -o test
neper -V test.tess -datacellcol group -print img
```
![img](https://user-images.githubusercontent.com/90365358/141294706-54f6c31c-178a-4a8a-b8d2-e5b33f981089.png)
But the grid file that was built didn't display correctly in ABAQUS
```
neper -M test.tess -elset "group:group" -format inp
```
![image](https://user-images.githubusercontent.com/90365358/141297803-ea2747e5-5b88-4e43-bade-a02dfb665acd.png)

Thank you and best regards!

**Comment by smallrain1996** [comment]
Also I have a similar question in the "FEPX Reference Manual... /examples/5_externa", Fepx requires the user to provide a simulation.phase file when simulating multiphase materials, obviously I could write a simple script myself based on the contents of the *group in the .tess, but could this be implemented in the -T module so that the user can easily get the .phase file required by Fepx, Thank you very much

**Reply by rquey** [reply]
Good point. This is a job for -M. Now available in version `4.2.1-13`:
```
neper -M test.tess -format msh,phase

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.2.1-13
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 8 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2021, and GNU GPL'd, by Romain Quey.
Info   : Loading initialization file `/home/rquey/.neperrc'...
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] -gmsh /home/rquey/bin/gmsh -tmp /home2/tmp
Info   : [com line] test.tess -format msh,phase
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `test.tess'...
Info   :     [i] Parsed file `test.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.1077) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing... 100% (0.045|0.64/96%| 3%| 1%)
Info   :   - Fixing 2D-mesh pinches...
Info   :   - 3D meshing... 100% (0.43|0.73/85%|15%| 0%) 
Info   : Searching nsets and fasets...
Info   : Writing mesh results...
Info   :   - Preparing mesh...
Info   :   - Mesh properties:
Info   :     > Node number:     5194
Info   :     > Elt  number:    26598
Info   :     > Mesh volume:    1.000
Info   :   - Writing mesh...
Info   :     [o] Writing file `test.msh'...
Info   :     [o] Wrote file `test.msh'.
Info   :     [o] Writing file `test.phase'...
Info   :     [o] Wrote file `test.phase'.
Info   : Elapsed time: 31.582 secs.
========================================================================
```

**Comment by rquey** [comment, marked as answer]
>  But the grid file that was built didn't display correctly in ABAQUS

What do you call a grid file? As far as I understand, it is (in Abaqus world) used to import data defined on a regular grid, but this is different from a tess file (which is what your -T command generates).

> `neper -M test.tess -elset "group:group" -format inp`

I don't understand the meaning of `group:group` here. What are you trying to achieve?

Anyway, the `*group` information is not written to the Abaqus file at the moment. I don't know how this should be done. A possibility is for you to get the group in a file and use this data to write your inp file.
```
neper -T -n 100 -group  "id<=50?1:2" -o test -statcell group
```

**Reply by siwux** [reply]
Thank you very much for your answer. Here `group:group`  I hope to build a set of groups so that I can assign the same material parameters 
to grains belonging to the same group in ABAQUS. But it has an error when use the syntax `-elset`
